cmd/podman/machine/cp.go

```diff
@@ -0,0 +1,158 @@
+//go:build amd64 || arm64
+
+package machine
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strconv"
+
+	"github.com/containers/podman/v5/cmd/podman/registry"
+	"github.com/containers/podman/v5/libpod/events"
+	"github.com/containers/podman/v5/pkg/copy"
+	"github.com/containers/podman/v5/pkg/machine"
+	"github.com/containers/podman/v5/pkg/machine/define"
+	"github.com/containers/podman/v5/pkg/machine/env"
+	"github.com/containers/podman/v5/pkg/machine/vmconfigs"
+	"github.com/containers/podman/v5/pkg/specgen"
+	"github.com/spf13/cobra"
+)
+
+type cpOptions struct {
+	Quiet    bool
+	Machine  *vmconfigs.MachineConfig
+	IsSrc    bool
+	SrcPath  string
+	DestPath string
+}
+
+var (
+	cpCmd = &cobra.Command{
+		Use:               "cp [options] SRC_PATH DEST_PATH",
+		Short:             "Securely copy contents between the virtual machine",
+		Long:              "Securely copy files or directories between the virtual machine and your host",
+		PersistentPreRunE: machinePreRunE,
+		RunE:              cp,
+		Args:              cobra.ExactArgs(2),
+		Example:           `podman machine cp ~/ca.crt podman-machine-default:/etc/containers/certs.d/ca.crt`,
+		ValidArgsFunction: autocompleteMachineCp,
+	}
+
+	cpOpts = cpOptions{}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: cpCmd,
+		Parent:  machineCmd,
+	})
+
+	flags := cpCmd.Flags()
+	quietFlagName := "quiet"
+	flags.BoolVarP(&cpOpts.Quiet, quietFlagName, "q", false, "Suppress copy status output")
+}
+
+func cp(_ *cobra.Command, args []string) error {
+	var err error
+
+	srcMachine, srcPath, destMachine, destPath, err := copy.ParseSourceAndDestination(args[0], args[1])
+	if err != nil {
+		return err
+	}
+
+	// Passing an absolute windows path of the format <volume>:\<path> will cause
+	// `copy.ParseSourceAndDestination` to think the volume is a Machine. Check
+	// if the raw cmdline argument is a Windows host path.
+	if runtime.GOOS == "windows" {
+		if specgen.IsHostWinPath(args[0]) {
```

Comment:
I am not a fan of the runtime checks OS checks, in tests that is fine but otherwise I prefer the the build tag option.
That said I don't think we need that at all do we? IsHostWinPath() calls shouldResolveWinPaths() which should return false already and true on windows so the condition is already in there.

Reply:
Yes, you're right. Thank you for pointing that out.